cartography/intel/aws/ec2.py

```diff
@@ -111,6 +149,22 @@ def load_ec2_instances(session, data, region, current_aws_account_id, aws_update
     SET r.lastupdated = {aws_update_tag}
     """
 
+    ingest_key_pair = """
+    MERGE (keypair:KeyPair:EC2KeyPair{keyname: {KeyName}, region: {Region}})
+    ON CREATE SET keypair.firstseen = timestamp()
+    SET keypair.lastupdated = {aws_update_tag}
+    WITH keypair
+    MATCH (aa:AWSAccount{id: {AWS_ACCOUNT_ID}})
+    MERGE (aa)-[r:RESOURCE]->(keypair)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag}
+    with keypair
+    MATCH (instance:EC2Instance{instanceid: {InstanceId}})
```

Comment:
Is it possible to have a Keypair that is not associated with EC2Instance? If so, I recommend to decouple the KeyPair node creation and the link to EC2Instance into separate statement

Reply:
It is possible, and `load_ec2_key_pairs()` handles that. But I think the cleanest way to be aware of the association between and EC2Instance and an EC2KeyPair is from within `load_ec2_instances()`